forms/common-themes/src/main/resources/theme/base/login/login-update-profile.ftl

```diff
@@ -6,6 +6,15 @@
         ${msg("loginProfileTitle")}
     <#elseif section = "form">
         <form id="kc-update-profile-form" class="${properties.kcFormClass!}" action="${url.loginAction}" method="post">
+            <div class="${properties.kcFormGroupClass!} ${messagesPerField.printIfExists('username',properties.kcFormGroupErrorClass!)}" <#if !realm.editUsernameAllowed>style="display:none"</#if>>
```

Comment:
This should not hide the fields with css, instead the whole section should be omitted from the output when username can't be changed